src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -304,6 +311,17 @@ internal void SetCookieOption(Uri uri)
                 }
             }
 
+            private void SetClientCertificateOptions()
+            {
+                if (_clientCertProvider == null)
+                {
+                    return;
+                }
+                SetCurlOption(CURLoption.CURLOPT_SSLCERT, _clientCertProvider._certificateFile);
```

Comment:
When does libcurl read in this file, during the setopt call?  Or later on when the request is made? I'm wondering if rather than creating the file earlier and disposing it at the end of the request if we could instead just do all of that locally in this function (and/or in the callback you're adding), deleting the file immediately after we pass the path to it?